### PR TITLE
Always support ignore_malformed in the same way

### PR DESCRIPTION
**Pull request by nik9000**
This makes sure that all field types that support `ignore_malfored` do so in the same way.

Production changes:
* All mapper has an `ignoreMalformed` method that must return `true` if the field accepts the `ignore_malformed` mapping parameter was configured. It defaults to `false` because many fields either don't have a concept of "malformed" value or don't have the ability to ignore malformed values.
* Fix the `scaled_float` field to store it's field name in `_ignored` if it ignores any malfored values. This is how all other field mappers work.

Test changes:
* `MapperTestCase` forces subclasses to declare if their `supportIgnoreMalformed` or not.
* If `MapperTestCase` subclasses `supportIgnoreMalfored` they must define some `exampleMalformedValues`.
* `MapperTestCase` always grows three new tests:
  * One that creates a field without setting `ignore_malformed` and verifies that all `exampleMalformedValues` throw expected errors
  * On that explicitly configured `ignore_malformed` to false and, if `supportIgnoreMalformed` it verifies the errors again. If not `supportIgnoreMalformed` it verifies that the parameter is unknown.
  * On that explicitly configured `ignore_malformed` to true and, if `supportIgnoreMalformed` it verifies that parsing doesn't produce errors and correctly produces `_ignored`. If not `supportIgnoreMalformed` it verifies that the parameter is unknown.
* Moved some subclasesses of `MapperTestCase` from `internalClusterTests` to `tests`. This isn't strictly required but that's the right place for them.
* Remove many hand rolled malformed tests, making sure that all removed examples are moved into the `exampleMalformedValues`

### CODE REVIEW

**Comment by nik9000** [comment]
My ulterior motive here is that I can reuse this infrastructure when we support `ignore_malformed` with synthetic `_source`.

File: plugins/mapper-annotated-text/build.gradle [hunk 1]
```diff
@@ -9,23 +9,18 @@ import org.elasticsearch.gradle.internal.info.BuildParams
...
-apply plugin: 'elasticsearch.internal-cluster-test'
 
 esplugin {
   description 'The Mapper Annotated_text plugin adds support for text fields with markup used to inject annotation tokens into the index.'
   classname 'org.elasticsearch.index.mapper.annotatedtext.AnnotatedTextPlugin'
 }
 
 if (BuildParams.isSnapshotBuild() == false) {
-  tasks.named("internalClusterTest").configure {
+  tasks.named("test").configure {
```
Comment (romseygeek) [inline plugins/mapper-annotated-text/build.gradle:19]: This has been annoying me for *ages*, thanks for fixing

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java [hunk 2]
```diff
@@ -137,6 +138,151 @@ protected void assertAggregatableConsistency(MappedFieldType ft) {
...
+         * {@link MapperTestCase#minimalMapping}.
+         */
+        public ExampleMalformedValue mapping(CheckedConsumer<XContentBuilder, IOException> newMapping) {
+            return new ExampleMalformedValue(newMapping, value, exceptionMessageMatcher);
+        }
+
+        /**
+         * Match a string contained in the error message.
+         */
+        public ExampleMalformedValue matcher(String contains) {
```
Comment (romseygeek) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:189]: Maybe call this `errorMatches` or something like that?  `matcher` reads oddly to me

File: x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java [hunk 3]
```diff
@@ -156,21 +148,42 @@ public void testParseEmptyValue() throws Exception {
...
+            exampleMalformedValue(b -> b.startObject().field("min", 10.0).field("max", 50.0).field("value_count", 77.33).endObject())
+                .matcher("failed to parse field [field.value_count] of type [integer]")
         );
-        ParsedDocument doc = mapper.parse(source(b -> b.startObject("field").endObject()));
-        assertThat(doc.rootDoc().getField("field"), nullValue());
+    }
+
+    @Override
+    protected String[] ignoredFields() {
+        return new String[] { "field.value_count", "field.min", "field.max" };
```
Comment (romseygeek) [inline x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java:186]: This is interesting - the `_ignored` field is really for user inspection, so I'm not sure if this mapper should be recording all its subfields in it as they are implementation details.

**Review by romseygeek — APPROVED** [review]
This is great, thanks @nik9000 

I left one question around how aggregate metric double records ignored values, but we can handle that in a follow-up if it's not one with an obvious answer.

File: x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java (hunk 3 above)
Comment (nik9000) [reply x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java:186]: Yeah! I think that's worth discussing somewhere. I can file an issue.

Comment (nik9000) [reply x-pack/plugin/mapper-aggregate-metric/src/test/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapperTests.java:186]: #90572

File: plugins/mapper-annotated-text/build.gradle (hunk 1 above)
Comment (nik9000) [reply plugins/mapper-annotated-text/build.gradle:19]: :bow:

**Pull request merged**